app/views/admin/recovery_displays/_form.html.haml

```diff
@@ -1,76 +1,51 @@
-.form.form-horizontal#chosen-container
-  = form_for [:admin, @recovery_display] do |f|
-    - if @recovery_display.errors.any?
-      #alert-block
-        .alert.alert-error
-          %h4
-            Please fix the following 
-            = pluralize(@recovery_display.errors.count, "error")
-          %ul
-            - @recovery_display.errors.full_messages.each do |msg|
-              %li
-                = msg
-    - if @recovery_display.stolen_record_id.present?
-      = f.hidden_field :stolen_record_id, value: @recovery_display.stolen_record_id
-    .well.clearfix
-      .span6
-        .control-group
-          = f.label :quote_by, class: "control-label"
+= form_for [:admin, @recovery_display] do |f|
+  - if @recovery_display.errors.any?
+    = render partial: "/shared/errors", locals: { name: "Recovery Display", obj: @recovery_display }
+  - if @recovery_display.stolen_record_id.present?
+    = f.hidden_field :stolen_record_id, value: @recovery_display.stolen_record_id
+  .form-group
+    = f.label :quote_by
+    = f.text_field :quote_by, class: "form-control w-25"
+  .form-group
+    = f.label :link
+    = f.text_field :link, class: "form-control w-25"
+  - if @bike.present?
+    .form-group
+      %label
+        Bike url
+      %input.less-strong{ type: 'text', value: bike_url(@recovery_display.bike) }
+  .form-group
+    = f.label :date_recovered
+    - f.object.date_recovered ||= Time.now
+    = f.datetime_local_field :date_recovered, max: Time.now + 1.day, step: 60, required: true, class: 'form-control w-25 dateInputUpdateZone', "data-initialtime" => l(f.object.date_recovered, format: :convert_time)
+  .form-group
+    = f.label :image, "Recovery photo"
+    - if @recovery_display.image?
+      = image_tag(@recovery_display.image_url(:thumb))
+    - else
+      %p
+        No picture!
+    = f.file_field :image, class: "float-right"
```

Comment:
the file upload is too far over! you should use columns to take care of this problem.